### ISSUE TITLE
Mouse cursor goes out of control sometimes on surface laptop

### ISSUE THREAD

**Issue by shashwat623** [linux-surface/linux-surface#447]
Hey developers,
I am facing a huge problem with the custom surface kernel. Sometimes after logging back in from sleep mode, the cursor goes out of control and is uncontrollable. It keeps hovering in all directions and randomly clicks stuff. Only restarting Ubuntu solves this. It has started to occur very frequently now. What can I do to solve this?

**Comment by SamLukeYes** [comment]
Looks like phantom touch. You can have a try with the command `sudo iptsd-reset-sensor` next time it occurs. If this doesn't help, preventing your laptop from sleeping would be another workaround :rofl:

**Comment by shashwat623** [comment]
@SamLukeYes Yeah, that's what I'm doing as of now, preventing my laptop form going into sleep mode. It would be almost impossible to type in the above command once the malfunction happens as it starts clicking everywhere randomly very fast. Is this problem with Ubuntu in general, or is it specific to the custom linux-surface kernel? Is it only me or even other using the linux-surface kernel on surface devices facing this issue? Any permanent solutions ??

**Comment by SamLukeYes** [comment]
>  It would be almost impossible to type in the above command once the malfunction happens as it starts clicking everywhere randomly very fast.

Have you tried switching to another tty? A non-graphical tty shouldn't be affected by mouse or touchscreen input. You can switch between ttys by pressing `Ctrl + Alt + F1~7`. If `iptsd-reset-sensor` works, then your issue should be the phantom touch that has been discussed before. It's a known issue of the new ipts driver.

**Comment by edegner** [comment]
This problem also affects Debian Bullseye, so it's not just Ubuntu.

Linux surfacebook 5.15.14-surface #1 SMP Sat Jan 15 02:27:14 UTC 2022 x86_64 GNU/Linux

Switching to a tty doesn't change anything.  If gpm is running you'll see the mouse cursor moving around and clicking on random things in the tty also.  I've found that removing the keyboard, then putting the screen to sleep, then attaching the screen back to the keyboard, then waking it up, SOMETIMES will stop the random clicking, but not every time.  It usually starts happening at the worst times too, and has caused me to lose hours worth of work.  Rebooting does make it go away, but rebooting is rarely an option.

**Comment by RichardScharf** [comment]
I can reproduce this problem on my Surface Book 1

**OS:** Linux Mint 20.3 x86_64
**Kernel:** 5.17.7-surface
**CPU:** Intel i7-6600U (4) @ 3.400GHz

**Comment by chiassonba** [comment]
I can reproduce this problem on my Surface Book 1 also.

OS: Debian 11 Bullseye - amd64
Kernel: 5.18.16-surface
CPU: Intel Core i5-63000 CPU@2.4GHz x 2


I also have an SB3 on Canadian Multilingual base dock. I saw that it has not been tested for the SB3, I would offer up to test it on that model, though I feel that this issue will also erupt.

If you do not ask me to do this, I'm just not planning to out of the kindness of my heart. I have 2 surface books, one is for me (The one with Debian) and the one for work .....Wind-blows and in-house developed "software" that crashes every 10 minutes.

I can flip my personal SB1 I use and my work SB3 without issues. My work does not own my surface books, worry not about any sort of difficulties.

Cheers, 
Have a super Friday everyone!
